File: back/back/Olap.WebApi/Models/ModelDescription.cs
using Olap.Model.ModelBuilder;
using System.Collections.Generic;

namespace Olap.WebApi.Models
{
    public class ModelDescription : IModelDescription
    {
        public string ModelName { get; set; }
        public IEnumerable<RowDescription> Rows { get; set; }
        public IEnumerable<ColumnDescription> Columns { get; set; }

        IEnumerable<IColumnDescription> IModelDescription.Columns => this.Columns;

        IEnumerable<IRowDescription> IModelDescription.Rows => this.Rows;
    }



}

File: front/src/common/modelDescription/modelDescription.ts
import IColumnDescription, { ColumnDescription } from './columnDescription.ts';
import IRowDescription, { RowDescription } from './rowDescription.ts';

export interface IModelDescription {
    readonly modelName: string;

    readonly rows: Array<IRowDescription>;

    readonly columns: Array<IColumnDescription>;
}

export default class ModelDescription implements IModelDescription {
  constructor(modelName: string, rows: Array<RowDescription>, columns: Array<ColumnDescription>) {
    this.modelName = modelName;
    this.rows = rows;
    this.columns = columns;
  }

    readonly modelName: string;

    readonly rows: Array<RowDescription>;

    readonly columns: Array<ColumnDescription>;
}
